sunkit_image/offlimb.py

```diff
@@ -0,0 +1,489 @@
+)
+
+__all__ = [
+    "fit_polynomial_to_log_radial_intensity",
+    "calculate_fit_radial_intensity",
+    "normalize_fit_radial_intensity",
+    "intensity_enhance",
+    "nrgf",
+    "fnrgf"
+]
+
+
+def fit_polynomial_to_log_radial_intensity(radii, intensity, degree):
+    """
+    Fits a polynomial of degree, "degree" to the log of the radial intensity.
+
+    Parameters
+    ----------
+    radii : `astropy.units.Quantity`
+        The radii at which the fitted intensity is calculated, nominally in
+        units of solar radii.
+    intensity : `numpy.ndarray`
+        The 1D intensity array to fit the polynomial for.
+    degree : `int`
+        The degree of the polynomial.
+
+    Returns
+    -------
+    polynomial : `numpy.ndarray`
+        A polynomial of degree, "degree" that fits the log of the intensity
+        profile as a function of the radius.
+    """
+    return np.polyfit(radii.to(u.R_sun).value, np.log(intensity), degree)
+
+
+def calculate_fit_radial_intensity(radii, polynomial):
+    """
+    Calculates the fit value of the radial intensity at the values "radii". The function assumes
+    that the polynomial is the best fit to the observed log of the intensity as a function of
+    radius.
+
+    Parameters
+    ----------
+    radii : `astropy.units.Quantity`
+        The radii at which the fitted intensity is calculated, nominally in
+        units of solar radii.
+    polynomial : `numpy.ndarray`
+        A polynomial of degree "degree" that fits the log of the intensity
+        profile as a function of the radius.
+
+    Returns
+    -------
+    fitted intensity : `numpy.ndarray`
+        An array with the same shape as radii which expresses the fitted
+        intensity value.
+    """
+    return np.exp(np.poly1d(polynomial)(radii.to(u.R_sun).value))
+
+
+def normalize_fit_radial_intensity(radii, polynomial, normalization_radius):
+    """
+    Normalizes the fitted radial intensity to the value at the normalization radius. The function
+    assumes that the polynomial is the best fit to the observed log of the intensity as a function
+    of radius.
+
+    Parameters
+    ----------
+    radii : `astropy.units.Quantity`
+        The radii at which the fitted intensity is calculated, nominally in
+        units of solar radii.
+    polynomial : `numpy.ndarray`
+        A polynomial of degree "degree" that fits the log of the intensity
+        profile as a function of the radius.
+    normalization_radius : `astropy.units.Quantity`
+        The radius at which the fitted intensity value is normalized to.
+
+    Returns
+    -------
+    normalized intensity : `numpy.ndarray`
+        An array with the same shape as radii which expresses the fitted
+        intensity value normalized to its value at the normalization radius.
+    """
+    return calculate_fit_radial_intensity(
+        radii, polynomial
+    ) / calculate_fit_radial_intensity(normalization_radius, polynomial)
+
+
+def intensity_enhance(
```

Comment:
This function should have a different name.  The current name is not that meaningful.  The way this code works is to divide the original emission by a fit to a summary statistic of the radial intensity, taken over all angles.  The fit is found by fitting a polynomial (in radius) to the logarithm of the summary statistic of the radial intensity.  So the function name should reflect this.  The difference between this function and the NRGF is the fact that a fit is done.  NRGF does not do a fit.  So how about changing this to `nrgf_fit`?

Reply:
I am not very sure about the scientific terms involved but it sure seems like what you are saying is correct. The name should be changed to something meaningful but according to me ```nrgf_fit``` also doesn't seem quite satisfactory. It creates an illusion that somehow this function is related to the NRGF function and paper.

Reply:
Yes, I agree.  How about `normalized_intensity` ?  That is sufficiently bland, and is a bit more descriptive than `intensity_enhance`.

Reply:
Yes that is much better